config/routes.rb

```diff
@@ -1,20 +1,62 @@
 Rails.application.routes.draw do
   namespace :api, defaults: { format: :json } do
     namespace :v1 do
-      get "/merchants/random", to: "random_merchants#show"
-      get "/merchants/find", to: "finds_merchants#show"
-      get "/merchants/find_all", to: "finds_merchants#index"
+      namespace :merchants do
+        get "/random", to: "random_merchants#show"
+        get "/find", to: "finds_merchants#show"
+        get "/find_all", to: "finds_merchants#index"
+      end
       resources :merchants, only: [:index, :show] do
-        resources :items, only: [:index]
-      end
-      get "/customers/random", to: "random_customers#show"
-      get "/customers/find", to: "finds_customers#show"
-      get "/customers/find_all", to: "finds_customers#index"
-      resources :customers, only: [:index, :show]
-      get "/transactions/random", to: "random_transactions#show"
-      get "/transactions/find", to: "finds_transactions#show"
-      get "/transactions/find_all", to: "finds_transactions#index"
-      resources :transactions, only: [:index, :show]
+        resources :items, only: [:index], :controller => "merchants/merchant_items"
+        resources :invoices, only: [:index], :controller => "merchants/merchant_invoices"
+      end
+      namespace :customers do
+        get "/random", to: "random_customers#show"
+        get "/find", to: "finds_customers#show"
+        get "/find_all", to: "finds_customers#index"
+      end
+      resources :customers, only: [:index, :show] do
+        resources :invoices, only: [:index], :controller => "customers/customer_invoices"
+        resources :transactions, only: [:index], :controller => "customers/customer_transactions"
+      end
+      namespace :items do
+        get "/random", to: "random_items#show"
```

Comment:
since we're namespaced within items (or in general whichever route you've defined), then we probably don't need to name the controller with `items` in it